---
title: C28105 | Microsoft-Dokumentation
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
caps.latest.revision: 5
author: mikeblome
ms.author: mblome
manager: jillfra
ms.openlocfilehash: 78fefc8101db0585d4e7fb741dd00ff9473698c8
ms.sourcegitcommit: 94b3a052fb1229c7e7f8804b09c1d403385c7630
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 04/23/2019
ms.locfileid: "68156399"
---
# <a name="c28105"></a>C28105
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Warnung C28105: Verlust von Ressource aufgrund einer Ausnahme  
  
 Die angegebene Ressource ist nicht freigegeben werden, wenn eine Ausnahme ausgelöst wird. Die Anweisung, die vom Pfad angegebenen kann es sich um eine Ausnahme ausgelöst. Diese Warnung ist ähnlich wie in "Warnung" [C28103](../code-quality/c28103.md), außer dass in diesem Fall eine Ausnahme beteiligt ist.  
  
## <a name="example"></a>Beispiel  
 Im folgenden Codebeispiel wird diese Warnung generiert:  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 Im folgenden Codebeispiel wird diese Warnung vermieden:  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```
